tiledb/tests/test_query_condition.py

```diff
@@ -67,3 +100,29 @@ def test_ints_floats_bytestrings(self):
             result = A.query(attr_cond=qc, attrs=["S"], use_arrow=False).df[:]
             assert len(result["S"]) == 1
             assert result["S"][0] == b"c"
+
+    def test_check_attrs(self, input_array_UIDS):
+        with tiledb.open(input_array_UIDS) as A:
+            qc = tiledb.QueryCondition("U < 0.1")
+            result = A.query(attr_cond=qc, attrs=["U"]).df[:]
+            assert all(result["U"] < 0.1)
+
+            qc = tiledb.QueryCondition("U < 1.0")
+            result = A.query(attr_cond=qc, attrs=["U"]).df[:]
+            assert all(result["U"] < 1.0)
+
+            qc = tiledb.QueryCondition("U < '1'")
```

Comment:
How does this work - is it casting the string to an int somewhere?

(Kind of seems like it should be an error to me, unless I'm misunderstanding what is happening here)

Reply:
https://github.com/TileDB-Inc/TileDB-Py/pull/589/files/85713a41f182faa67090bcc20490c36a9aaf9b52#diff-bc89368eea7e3bcb8e4e64c38e4b2caf1f1ae9ddbf27846dd9363e56d1114367R149-R150

NumPy is doing the casting here to turn this string into an int. In the case that the string can't be casting to an int, it errors out.